plugins/entando-plugin-jpdigital-exchange/src/test/java/org/entando/entando/web/digitalexchange/install/DigitalExchangeInstallResourceIntegrationTest.java

```diff
@@ -166,51 +179,137 @@ public void shouldInstallWidget() throws Exception {
 
         String componentCode = "de_test_widget";
 
-        installAndCheckForCompletion(componentCode);
+        try {
+            installAndCheckForCompletion(componentCode);
+
+            assertThat(widgetService.getWidget(componentCode)).isNotNull();
+
+
+        } finally {
+
+            try {
+                if (widgetService.getWidget(componentCode) != null) {
+                    widgetService.removeWidget(componentCode);
+                }
+
+            } catch (RestRourceNotFoundException ignored) {
+
+            }
+
+        }
 
-        assertThat(widgetService.getWidget(componentCode)).isNotNull();
 
-        widgetService.removeWidget(componentCode);
     }
 
+
     @Test
     public void shouldInstallPageModel() throws Exception {
 
         String componentCode = "de_test_page_model";
+        String componentAssociatedLabel = "DE_TEST_LABEL";
 
-        installAndCheckForCompletion(componentCode);
+        try {
 
-        assertThat(pageModelService.getPageModel(componentCode)).isNotNull();
-        assertThat(i18nManager.getLabel("DE_TEST_LABEL", "en")).isEqualTo("Test label DE");
-        assertThat(storageManager.exists("components/de_test_page_model/test.css", false)).isTrue();
+            installAndCheckForCompletion(componentCode);
+
+            assertThat(pageModelService.getPageModel(componentCode)).isNotNull();
+            assertThat(i18nManager.getLabel("DE_TEST_LABEL", "en")).isEqualTo("Test label DE");
+            assertThat(storageManager.exists("components/de_test_page_model/test.css", false)).isTrue();
+
+        } finally {
+
+            if (pageModelService.getPageModel(componentCode) != null) {
+                pageModelService.removePageModel(componentCode);
+            }
+
+            if (labelService.getLabelGroup(componentAssociatedLabel) != null) {
```

Comment:
Same here, maybe we could think about injecting managers instead of services, because managers returns null if the objects are not found.